cashier-google-play-billing/src/main/java/com/getkeepsafe/cashier/billing/GooglePlayBillingVendor.java

```diff
@@ -0,0 +1,500 @@
+    }
+
+    @Override
+    public synchronized void initialize(Context context, InitializationListener listener) {
+        Preconditions.checkNotNull(context, "Cannot initialize with null context");
+        Preconditions.checkNotNull(listener, "Cannot initialize with null initialization listener");
+
+        if (available()) {
+            listener.initialized();
+            return;
+        }
+
+        initializationListeners.add(listener);
+
+        if (!initializing) {
+            initializing = true;
+            logSafely("Initializing Google Play Billing API...");
+            available = api.initialize(context, this, this, logger);
+        }
+
+        if (!available) {
+            initializationListeners.remove(listener);
+            listener.unavailable();
+        }
+    }
+
+    @Override
+    public synchronized void initialized(boolean success) {
+        logSafely("Initialized: success = " + success);
+        if (!success) {
+            logAndDisable("Could not create Google Play Billing instance");
+            return;
+        }
+
+        try {
+            canPurchaseItems =
+                    api.isBillingSupported(SkuType.INAPP) == BillingResponse.OK;
+
+            canSubscribe =
+                    api.isBillingSupported(SkuType.SUBS) == BillingResponse.OK;
+
+            available = canPurchaseItems || canSubscribe;
+            logSafely("Connected to service and it is " + (available ? "available" : "not available"));
+            initializing = false;
+
+            for (InitializationListener listener : initializationListeners) {
+                listener.initialized();
+            }
+        } catch (Exception error) {
+            logAndDisable(Log.getStackTraceString(error));
+        }
+    }
+
+    @Override
+    public void disconnected() {
+        logAndDisable("Disconnected from Google Play Billing service.");
+    }
+
+    @Override
+    public void dispose(Context context) {
+        logSafely("Disposing Google Play Billing vendor...");
+        api.dispose();
+        available = false;
+        initializationListeners.clear();
+    }
+
+    @Override
+    public synchronized void purchase(Activity activity, Product product, String developerPayload, PurchaseListener listener) {
+        Preconditions.checkNotNull(activity, "Activity is null.");
+        Preconditions.checkNotNull(product, "Product is null.");
+        Preconditions.checkNotNull(listener, "Purchase listener is null.");
+        throwIfUninitialized();
+
+        if (pendingProduct != null) {
+            throw new RuntimeException("Cannot purchase product while another purchase is in progress!");
+        }
+
+        // NOTE: Developer payload is not supported with Google Play Billing
+        // https://issuetracker.google.com/issues/63381481
+        if (developerPayload != null && developerPayload.length() > 0) {
+            throw new RuntimeException("Developer payload is not supported in Google Play Billing!");
```

Comment:
I'm not sure if setting the developer payload should crash the client code. A log warning might be enough?

Reply:
If developer payload was used with iab implementation, the client code may rely on it to verify purchase. Since we can't change cashier's interface, I wanted to make sure that after upgrade this change in behaviour doesn't go unnoticed. I believe it is better than silently failing to return developer payload in purchase. I doubt that log would be enough.

Reply:
I understand. If this is the thinking, I suggest we throw an exemption in calling `GooglePlayBillingPurchase#developerPayload` as well so that if the client migrated to this new vendor and was previously doing verification with the developer payload should be aware that it is not supported. Please add documentation to that method as well with link to the issue similar to `GooglePlayBillingPurchase L:31`. I would also suggest to document this migration in the README.md.